src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php

```diff
@@ -480,9 +480,11 @@ private function updateSchema(): void
 
         $schemaManager = $this->createSchemaManager();
         $comparator = $this->createComparator($schemaManager);
-        $schemaDiff = $this->compareSchemas($comparator, $schemaManager->createSchema(), $this->getSchema());
+        $schemaDiff = $this->compareSchemas($comparator, method_exists($schemaManager, 'introspectSchema') ? $schemaManager->introspectSchema() : $schemaManager->createSchema(), $this->getSchema());
```

Comment:
@nicolas-grekas actually this change deletes all existing tables in the database 😬

Reply:
@xabbuh yeah, I also thought so, but when I just reverted this code block (not the entire file) it continued to work as expected. But yes, I'll create a follow-up ticket with a reproducible case. Thanks so far 👍😊